### PR TITLE
Doc: Automatic generation of docs for collection plugins

### LINKED ISSUE #3034: Documentation: Make sure Ansible plugins are properly documented and use this to keep our doc up to date
### Enhancement summary

Probably multiples PRs to go through to achieve this

### Which component of AVD is impacted

plugins, others

### Use case example

when updating yaml_template_to_fact inputs, the avd.sh doc is not updated automatically which can lead to misleading doc

### Describe the solution you would like

go through doc of all plugins in order to leverage this for avd doc

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Adding Ansible documentation strings to collection plugins and building docs automatically.

Fixes #3034 

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/filter/add_md_toc.py [hunk 1]
```diff
@@ -18,6 +18,59 @@
...
+requirements:
+  - md_toc
+short_description: Parse the input MarkDown and add a Table of Contents between the toc_markers.
+description:
+  - The filter is used in M(arista.avd.eos_designs) to create a table of contents for Fabric Documentation.
+  - The filter is also used in M(arista.avd.eos_cli_config_gen) to create a table of contents for Device Documentation.
+positional: _input
+options:
+  _input:
+    description: MarkDown which will be processed.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/add_md_toc.py]:
```suggestion
    description: Markdown to process.
```

File: ansible_collections/arista/avd/plugins/filter/add_md_toc.py [hunk 2]
```diff
@@ -18,6 +18,59 @@
...
+    type: string
+    required: true
+  skip_lines:
+    type: integer
+    description: Skip first x lines when parsing the input MarkDown.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/add_md_toc.py]:
```suggestion
    description: Skip the first x lines when parsing the input Markdown.
```

File: ansible_collections/arista/avd/plugins/filter/add_md_toc.py [hunk 3]
```diff
@@ -18,6 +18,59 @@
 except ImportError:
     HAS_MD_TOC = False
 
+DOCUMENTATION = r"""
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/add_md_toc.py:24]: Any reason to case Markdown to MarkDown?

File: ansible_collections/arista/avd/plugins/filter/add_md_toc.py [hunk 4]
```diff
@@ -18,6 +18,59 @@
...
+    type: integer
+    description: Skip first x lines when parsing the input MarkDown.
+    default: 0
+  toc_levels:
+    type: integer
+    description: How many levels of headings will be included in the TOC.
+    default: 3
+  toc_marker:
+    type: string
+    description: TOC will be inserted or updated between two of these markers in the input MarkDown.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/add_md_toc.py]:
```suggestion
    description: TOC will be inserted or updated between two of markers in the input Markdown.
```

File: ansible_collections/arista/avd/plugins/filter/add_md_toc.py [hunk 5]
```diff
@@ -18,6 +18,59 @@
...
 
+DOCUMENTATION = r"""
+---
+name: add_md_toc
+collection: arista.avd
+author: Arista Ansible Team (@aristanetworks)
+version_added: "3.2"
+requirements:
+  - md_toc
+short_description: Parse the input MarkDown and add a Table of Contents between the toc_markers.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/add_md_toc.py]: short_description doesn't seem to match the preview site.

File: ansible_collections/arista/avd/plugins/filter/convert_dicts.py [hunk 6]
```diff
@@ -8,6 +8,81 @@
...
 
+DOCUMENTATION = r"""
+---
+name: convert_dicts
+collection: arista.avd
+author: Arista Ansible Team (@aristanetworks)
+version_added: "2.0"
+short_description: Convert a dictionary containing nested dictionaries to a list of dictionaries.
+description:
+  - The filter inserts the outer dictionary keys into each list item using the primary_key `name` (key name is
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/convert_dicts.py]:
```suggestion
  - The filter inserts the outer dictionary keys into each list item using the primary_key `name` (the key name is
```

File: ansible_collections/arista/avd/plugins/filter/convert_dicts.py [hunk 7]
```diff
@@ -8,6 +8,81 @@
...
+    configurable) and if there is a non-dictionary value,it inserts this value to
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/convert_dicts.py]:
```suggestion
    configurable), and if there is a non-dictionary value, it inserts this value to
```

File: ansible_collections/arista/avd/plugins/filter/convert_dicts.py [hunk 8]
```diff
@@ -8,6 +8,81 @@
...
+    secondary key (key name is configurable), if I(secondary_key) is provided.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/convert_dicts.py:21]:
```suggestion
    the secondary key (the key name is configurable) if a `secondary_key` is provided.
```

File: ansible_collections/arista/avd/plugins/filter/convert_dicts.py [hunk 9]
```diff
@@ -8,6 +8,81 @@
...
+  - This filter is intended for seemless data model migration from dictionaries to lists.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/convert_dicts.py]:
```suggestion
  - This filter is intended for seamless data model migration from dictionaries to lists.
```

File: ansible_collections/arista/avd/plugins/filter/convert_dicts.py [hunk 10]
```diff
@@ -8,6 +8,81 @@
...
+  - The filter can improve Ansible's processing performance when dealing with large dictionaries by converting them to lists of dictionaries.
+  - Note - if there is a non-dictionary value with no secondary key provided, it will pass through untouched.
+
+positional: _input
+options:
+  _input:
+    type: any
+    description: Dictionary to convert - returned untouched if it is not a nested dictionary/list.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/convert_dicts.py]:
```suggestion
    description: Dictionary to convert - returned untouched if not a nested dictionary/list.
```

File: ansible_collections/arista/avd/plugins/filter/convert_dicts.py [hunk 11]
```diff
@@ -8,6 +8,81 @@
...
+    required: true
+  primary_key:
+    type: string
+    description: Name of primary key used when inserting outer dictionary keys into items.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/convert_dicts.py]:
```suggestion
    description: Name of the primary key used when inserting outer dictionary keys into items.
```

File: ansible_collections/arista/avd/plugins/filter/convert_dicts.py [hunk 12]
```diff
@@ -8,6 +8,81 @@
...
+    default: name
+  secondary_key:
+    type: string
+    description: Name of secondary key used when inserting dictionary values which are list into items.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/convert_dicts.py]:
```suggestion
    description: Name of the secondary key used when inserting dictionary values which are list into items.
```

File: ansible_collections/arista/avd/plugins/filter/default.py [hunk 13]
```diff
@@ -7,6 +7,42 @@
 
 from jinja2.runtime import Undefined
 
+DOCUMENTATION = r"""
+---
+name: default
+collection: arista.avd
+author: Arista Ansible Team (@aristanetworks)
+version_added: "2.0"
+short_description: Returns input value if defined and is not none. Otherwise return default value.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/default.py]:
```suggestion
short_description: Returns input value if defined and is not none. Otherwise, return default value.
```

File: ansible_collections/arista/avd/plugins/filter/default.py [hunk 14]
```diff
@@ -7,6 +7,42 @@
...
+  As a last resort, the filter will return C(None).
+positional: _input
+options:
+  _input:
+    description: Default value to check. Will be returned untouched if C(not undefined) and C(not None).
+    type: any
+    required: true
+  default_values:
+    type: any
+    description: One or more default values which will be tested one by one and the first valid value will be used.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/default.py]:
```suggestion
    description: One or more default values will be tested individually, and the first valid value will be used.
```

File: ansible_collections/arista/avd/plugins/filter/encrypt.py [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+
+DOCUMENTATION = r"""
+---
+name: encrypt
+collection: arista.avd
+author: Arista Ansible Team (@aristanetworks)
+version_added: "3.8.0"
+short_description: Encrypt supported EOS passwords
+description:
+  - The filter is used to encrypt clear text password into EOS passwords.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/encrypt.py]:
```suggestion
  - The filter encrypts a clear text password into EOS passwords.
```

File: ansible_collections/arista/avd/plugins/filter/encrypt.py [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+  - To be used in conjunction with Ansible Vault to be able to load a password and have it encrypted on the fly by AVD in C(eos_designs).
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/encrypt.py]:
```suggestion
  - To be used with Ansible Vault to load a password and have it encrypted on the fly by AVD in C(eos_designs).
```

File: ansible_collections/arista/avd/plugins/filter/encrypt.py [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+      - Type of password to encrypt.
+      - C(bgp) and C(ospf_simple) requires the I(password) and I(key) inputs.
+      - C(ospf_message_digest) requires the I(password), I(key), I(hash_algorithm), I(key_id) inputs.
+    choices: ["bgp", "ospf_simple", "ospf_message_digest"]
+    required: true
+  key:
+    type: string
+    description:
+      - Encryption key. The value depends on the type of password.
+      - For BGP passwords the key is the Neighbor IP or the BGP Peer Group Name in EOS.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/encrypt.py]:
```suggestion
      - For BGP passwords, the key is the Neighbor IP or the BGP Peer Group Name in EOS.
```

File: ansible_collections/arista/avd/plugins/filter/encrypt.py [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+      - For OSPF passwords the key is the interface name (e.g., C(Ethernet1)).
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/encrypt.py]:
```suggestion
      - For OSPF passwords, the key is the interface name (e.g., C(Ethernet1)).
```

File: ansible_collections/arista/avd/plugins/filter/is_in_filter.py [hunk 19]
```diff
@@ -5,6 +5,45 @@
...
 
+DOCUMENTATION = r"""
+---
+name: is_in_filter
+collection: arista.avd
+author: Arista Ansible Team (@aristanetworks)
+version_added: "1.1"
+short_description: Returns V(True) if the input hostname matches the I(hostname_filter).
+description:
+  - The filter matches if any of the filter strings are found in the given input hostname.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/is_in_filter.py]:
```suggestion
  - The filter matches if any filter strings are found in the input hostname.
```

File: ansible_collections/arista/avd/plugins/filter/list_compress.py [hunk 20]
```diff
@@ -9,6 +9,37 @@
...
 
+DOCUMENTATION = r"""
+---
+name: list_compress
+collection: arista.avd
+author: Arista Ansible Team (@aristanetworks)
+version_added: "1.0"
+short_description: Compress a list of integers to a range string.
+description:
+  - Provides the capability to generate a string representing ranges of VLANs or VNIs
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/list_compress.py]:
```suggestion
  - Provides the capability to generate a string representing ranges of VLANs or VNIs.
```

File: ansible_collections/arista/avd/plugins/filter/markdown_rendering.py [hunk 21]
```diff
@@ -3,6 +3,37 @@
...
+  rendering:
+    description: Markdown Flavor to use for Emoji rendering. Only "github" is supported.
+    type: string
+    required: true
+"""
+
+RETURN = r"""
+---
+_value:
+  description: Input value or github markdown emoji code.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/markdown_rendering.py]:
```suggestion
  description: Input value or GitHub Markdown emoji code.
```

File: ansible_collections/arista/avd/plugins/filter/natural_sort.py [hunk 22]
```diff
@@ -10,6 +10,47 @@
...
+DOCUMENTATION = r"""
+---
+name: natural_sort
+collection: arista.avd
+author: Arista Ansible Team (@aristanetworks)
+version_added: "1.0.0"
+short_description: Sort an input list with natural sorting.
+description:
+  - |-
+      Provides the capabilities to sort a list or a dictionary of integers and/or strings that contain alphanumeric characters naturally.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/natural_sort.py]:
```suggestion
      Provides the capability to sort a list or a dictionary of integers and strings that contain alphanumeric characters naturally.
```

File: ansible_collections/arista/avd/plugins/filter/snmp_hash.py [hunk 23]
```diff
@@ -8,6 +8,51 @@
...
+collection: arista.avd
+author: Arista Ansible Team (@aristanetworks)
+version_added: "3.6.0"
+short_description: Compute localized SNMP passphrases
+description:
+  - Key localization as described in L(RFC 2574 section 2.6,https://www.rfc-editor.org/rfc/rfc2574.html#section-2.6)
+positional: _input
+options:
+  _input:
+    description: Dictionary with snmp passphrase details.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/snmp_hash.py]:
```suggestion
    description: Dictionary with SNMP passphrase details.
```

File: ansible_collections/arista/avd/plugins/filter/snmp_hash.py [hunk 24]
```diff
@@ -8,6 +8,51 @@
...
+    type: dictionary
+    required: true
+    suboptions:
+      passphrase:
+        type: string
+        required: true
+        description: The passphrase to localize, if I(priv) is not set it is the "auth" passphrase else it is the "priv" passphrase.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/filter/snmp_hash.py]: Hard to read sentence. Consider rewriting

File: ansible_collections/arista/avd/plugins/modules/eos_designs_facts.py [hunk 25]
```diff
@@ -33,9 +33,9 @@
     and python module in `arista.avd.eos_designs` to generate the `structured_configuration`.
 options:
   template_output:
-    description: |
-      If true the output data will be run through another jinja2 rendering before returning.
-      This is to resolve any input values with inline jinja using variables/facts set by the input templates.
+    description:
+      - If true the output data will be run through another jinja2 rendering before returning.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/modules/eos_designs_facts.py]:
```suggestion
      - If true, the output data will be run through another jinja2 rendering before returning.
```

File: ansible_collections/arista/avd/plugins/modules/eos_designs_structured_config.py [hunk 26]
```diff
@@ -49,26 +49,26 @@
...
   mode:
     description: File mode (ex. 0664) for dest file. See 'ansible.builtin.copy' module for details.
     required: false
     type: str
   template_output:
-    description: |
-      If true the output data will be run through another jinja2 rendering before returning.
-      This is to resolve any input values with inline jinja using variables/facts set by the input templates.
+    description:
+      - If true the output data will be run through another jinja2 rendering before returning.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/modules/eos_designs_structured_config.py]:
```suggestion
      - If true, the output data will be run through another jinja2 rendering before returning.
```

File: ansible_collections/arista/avd/plugins/modules/verify_requirements.py [hunk 27]
```diff
@@ -18,7 +18,7 @@
 version_added: "4.0.0"
 author: Arista Ansible Team (@aristanetworks)
 short_description: Verify Python requirements when running AVD
-description:
+description: |-
   The `arista.avd.verify_requirements` module is an Ansible Action Plugin providing the following capabilities
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/modules/verify_requirements.py]:
```suggestion
  The `arista.avd.verify_requirements` module is an Ansible Action Plugin providing the following capabilities:
```

File: ansible_collections/arista/avd/plugins/test/defined.py [hunk 28]
```diff
@@ -26,6 +26,60 @@
...
+name: defined
+collection: arista.avd
+author: Arista Ansible Team (@aristanetworks)
+version_added: "2.0"
+short_description: Test if the value is not C(Undefined) or C(None).
+description:
+  - The M(arista.avd.defined) test returns C(False) if the passed value is C(Undefined) or C(None). Else it will return C(True).
+  - The M(arista.avd.defined) test also accepts an optional I(test_value) argument to test if the value equals this.
+  - The optional I(var_type) argument can also be used to test if the variable is of the expected type.
+  - Optionally the test can emit warnings or errors if the test fails.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/test/defined.py]:
```suggestion
  - Optionally, the test can emit warnings or errors if the test fails.
```

File: ansible_collections/arista/avd/plugins/test/defined.py [hunk 29]
```diff
@@ -26,6 +26,60 @@
...
+  - Compared to the builtin C(is defined) test, this test will also test for C(None) and can even test for a specific value or class.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/test/defined.py]:
```suggestion
  - Compared to the built-in C(is defined) test, this test will also test for C(None) and can even test for a specific value or class.
```

File: ansible_collections/arista/avd/plugins/test/defined.py [hunk 30]
```diff
@@ -26,6 +26,60 @@
...
+  test_value:
+    description: Value to match for in addition to defined and not none
+    default: None
+  var_type:
+    type: string
+    description: Type or Class to test for
+    choices: ['float', 'int', 'str', 'list', 'dict', 'tuple', 'bool']
+  fail_action:
+    type: string
+    description: Optional action if test fails to emit a Warning or Error
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/test/defined.py]:
```suggestion
    description: Optional action if a test fails to emit a Warning or Error
```

File: ansible_collections/arista/avd/plugins/test/defined.py [hunk 31]
```diff
@@ -26,6 +26,60 @@
...
+    choices: ['warning', 'error']
+  var_name:
+    type: string
+    description: Optional string to use as variable name in warning or error messages
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/test/defined.py]:
```suggestion
    description: Optional string to use as a variable name in warning or error messages
```

File: ansible_collections/arista/avd/plugins/filter/add_md_toc.py [hunk 32]
```diff
@@ -18,6 +18,59 @@
...
+  copy:
+    content: "{{ lookup('template','documentation/fabric-documentation.j2') | arista.avd.add_md_toc(skip_lines=3) }}"
+    dest: "{{ fabric_dir }}/{{ fabric_name }}-documentation.md"
+    mode: 0664
+"""
+
+RETURN = r"""
+---
+_value:
+  description: MarkDown with TOC inserted between the toc_markers.
```
Comment (Shivani-chourasiya) [inline ansible_collections/arista/avd/plugins/filter/add_md_toc.py]:
```suggestion
  description: Markdown with TOC inserted between the toc_markers.
```

File: ansible_collections/arista/avd/plugins/filter/convert_dicts.py (hunk 8 above)
Comment (Shivani-chourasiya) [inline ansible_collections/arista/avd/plugins/filter/convert_dicts.py]:
```suggestion
    secondary key (the key name is configurable), if I(secondary_key) is provided.
```

File: ansible_collections/arista/avd/plugins/filter/default.py [hunk 33]
```diff
@@ -7,6 +7,42 @@
...
+
+EXAMPLES = r"""
+---
+myvalue: "{{ variable | arista.avd.default(default_value_1, default_value_2) }}"
+"""
+
+RETURN = r"""
+---
+_value:
+  description: Input value if C(not undefined) and C(not None). Otherwise return first defined default value or C(None).
```
Comment (Shivani-chourasiya) [inline ansible_collections/arista/avd/plugins/filter/default.py]:
```suggestion
  description: Input value if C(not undefined) and C(not None). Otherwise, return the first defined default value or C(None).
```

File: ansible_collections/arista/avd/plugins/modules/yaml_templates_to_facts.py [hunk 34]
```diff
@@ -81,19 +74,19 @@
...
   mode:
     description: File mode (ex. 0664) for dest file. See 'ansible.builtin.copy' module for details.
     required: false
     type: str
   template_output:
-    description: |
-      If true the output data will be run through another jinja2 rendering before returning.
-      This is to resolve any input values with inline jinja using variables/facts set by the input templates.
+    description:
+      - If true the output data will be run through another jinja2 rendering before returning.
```
Comment (Shivani-chourasiya) [inline ansible_collections/arista/avd/plugins/modules/yaml_templates_to_facts.py]:
```suggestion
      - If true, the output data will be run through another jinja2 rendering before returning.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 35]
```diff
@@ -2,84 +2,67 @@
...
-    requirements:
-        - This plugin should run at the `inventory` stage (default) before all other variable plugins, to inject the variables before any group and host vars.
+DOCUMENTATION = r"""
+---
+name: global_vars
+author: Arista Ansible Team (@aristanetworks)
+version_added: "4.0.0"
+short_description: Variable plugins to allow loading global_vars with less precedence than group_vars or host_vars
+requirements:
+  - This plugin should run at the `inventory` stage (default) before all other variable plugins, to inject the variables before any group and host vars.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
  - This plugin should run at the `inventory` stage (default) before all other variable plugins to inject the variables before any group and host vars.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 36]
```diff
@@ -2,84 +2,67 @@
...
+description:
+  - Loads variables from variable files specified in ansible.cfg or in the environment variable.
+  - Assign the loaded variables to the 'all' inventory group.
+  - Files are restricted by extension to one of .yaml, .json, .yml or no extension.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
  - Files are restricted by extension to one of .yaml, .json, .yml, or no extension.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 37]
```diff
@@ -2,84 +2,67 @@
...
+  - Hidden files (starting with '.') and backup files (ending with '~') files are ignored.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
  - Hidden files (starting with '.') and backup files (ending with '~') are ignored.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 38]
```diff
@@ -2,84 +2,67 @@
...
-    [defaults]
-    vars_plugins_enabled = arista.avd.global_vars, host_group_vars
-    ```
-
-2. Run your playbook
-
-    ```shell
-    ARISTA_AVD_GLOBAL_VARS_PATHS=../relative/path/to/my/global/vars/file/or/dir ansible-playbook -i inventory.yml playbook.yml
-    ```
+      - List of relative paths, relative to the inventory file.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
      - List of relative paths relative to the inventory file.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 39]
```diff
@@ -2,84 +2,67 @@
...
+      - If path is a directory, all the valid files inside are loaded in alphabetical order.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
      - If path is a directory, all the valid files inside are loaded alphabetically.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 40]
```diff
@@ -2,84 +2,67 @@
...
+  stage:
+    default: inventory
+    choices: ["inventory"]
+    description:
+      - The stage during which executing the plugin. It could be 'inventory' or 'task'
+      - Given the expected usage of this plugin at the beginning of the run. It is hardcoded to 'inventory'
+  _valid_extensions:
+    default: [".yml", ".yaml", ".json"]
+    description:
+      - Check all of these extensions when looking for 'variable' files which should be YAML or JSON or vaulted versions of these.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
      - Check all of these extensions when looking for 'variable' files, which should be YAML, JSON, or vaulted versions.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 41]
```diff
@@ -2,84 +2,67 @@
...
+      - This affects vars_files, include_vars, inventory and vars plugins among others.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
      - This affects vars_files, include_vars, inventory, and vars plugins, among others.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 42]
```diff
@@ -2,84 +2,67 @@
...
-    ```
+      - List of relative paths, relative to the inventory file.
+      - If path is a directory, all the valid files inside are loaded in alphabetical order.
+      - If the environment variable is set, it takes precedence over ansible.cfg.
+  stage:
+    default: inventory
+    choices: ["inventory"]
+    description:
+      - The stage during which executing the plugin. It could be 'inventory' or 'task'
+      - Given the expected usage of this plugin at the beginning of the run. It is hardcoded to 'inventory'
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
      - Given the expected usage of this plugin at the beginning of the run. It is hard-coded to 'inventory'
```

File: ansible_collections/arista/avd/plugins/filter/add_md_toc.py (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/filter/add_md_toc.py:24]: only because it was so already. Changing.

File: ansible_collections/arista/avd/plugins/filter/add_md_toc.py (hunk 5 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/filter/add_md_toc.py]: not understood

File: ansible_collections/arista/avd/plugins/filter/snmp_hash.py (hunk 24 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/filter/snmp_hash.py]: better?

File: ansible_collections/arista/avd/plugins/filter/add_md_toc.py (hunk 5 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/filter/add_md_toc.py]: Turned out that another module had `add_md_toc` as the name as well, leading to duplicate docs and confusing this. Fixed it now.

File: ansible_collections/arista/avd/docs/plugins/Filter plugins/convert_dicts.md [hunk 43]
```diff
@@ -0,0 +1,76 @@
+# convert_dicts
+
+!!! note
+    Always use the FQCN (Fully Qualified Collection Name) `arista.avd.convert_dicts` when using this plugin.
+
+Convert a dictionary containing nested dictionaries to a list of dictionaries\.
+
+## Synopsis
+
+The filter inserts the outer dictionary keys into each list item using the primary\_key \`name\` \(the key name is configurable\)\, and if there is a non\-dictionary value\, it inserts this value to secondary key \(the key name is configurable\)\, if <em>secondary\_key</em> is provided\.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/docs/plugins/Filter plugins/convert_dicts.md]: Some of the `\` are present in the [preview site](https://ansible-avd--3061.org.readthedocs.build/en/3061/docs/plugins/Filter%20plugins/convert_dicts.html#synopsis)

File: ansible_collections/arista/avd/docs/plugins/Filter plugins/decrypt.md [hunk 44]
```diff
@@ -0,0 +1,48 @@
+# decrypt
+
+!!! note
+    Always use the FQCN (Fully Qualified Collection Name) `arista.avd.decrypt` when using this plugin.
+
+Decrypt supported EOS passwords
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/docs/plugins/Filter plugins/decrypt.md]:
```suggestion
Decrypt supported EOS passwords.
```

File: ansible_collections/arista/avd/docs/plugins/Filter plugins/decrypt.md [hunk 45]
```diff
@@ -0,0 +1,48 @@
...
+
+## Synopsis
+
+The filter is used to decrypt supported EOS passwords into clear text\.
+
+Note \- For now this filter only supports decryption from type <code>7</code> and not type <code>8a</code> for OSPF and BGP passwords\.
+
+## Parameters
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/docs/plugins/Filter plugins/decrypt.md:20]: The escape sequence is present in the [table](https://ansible-avd--3061.org.readthedocs.build/en/3061/docs/plugins/Filter%20plugins/decrypt.html#parameters).

File: ansible_collections/arista/avd/docs/plugins/Filter plugins/range_expand.md [hunk 46]
```diff
@@ -0,0 +1,76 @@
...
+- "{{ 'Ethernet1-2' | range_expand }}"
+# -> ["Ethernet1", "Ethernet2"]
+
+- "{{ 'Eth 3-5,7-8' | range_expand }}"
+# -> ["Eth 3", "Eth 4", "Eth 5", "Eth 7", "Eth 8"]
+
+- "{{ 'et2-6,po1-2' | range_expand }}"
+# -> ["et2", "et3", "et4", "et5", "et6", "po1", "po2"]
+
+- "{{ ['Ethernet1'] | range_expand }}"
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/docs/plugins/Filter plugins/range_expand.md:44]: Duplicate example to the first

File: ansible_collections/arista/avd/docs/plugins/Modules and action plugins/eos_designs_structured_config.md [hunk 47]
```diff
@@ -0,0 +1,59 @@
+# eos_designs_structured_config
+
+!!! note
+    Always use the FQCN (Fully Qualified Collection Name) `arista.avd.eos_designs_structured_config` when using this plugin.
+
+Generate AVD EOS Designs structured configuration
+
+## Synopsis
+
+Validates input variables according to eos\_designs schema
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/docs/plugins/Modules and action plugins/eos_designs_structured_config.md]: Consider making these sentences if not, then possible make them a list

File: ansible_collections/arista/avd/docs/plugins/Modules and action plugins/inventory_to_container.md [hunk 48]
```diff
@@ -0,0 +1,61 @@
+# inventory_to_container
+
+!!! note
+    Always use the FQCN (Fully Qualified Collection Name) `arista.avd.inventory_to_container` when using this plugin.
+
+Transform information from inventory to arista\.cvp collection
+
+## Synopsis
+
+Transform information from ansible inventory to be able to
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/docs/plugins/Modules and action plugins/inventory_to_container.md]: Rendering places these on individual lines even though its [one sentence](https://ansible-avd--3061.org.readthedocs.build/en/3061/docs/plugins/Modules%20and%20action%20plugins/inventory_to_container.html#synopsis).

File: ansible_collections/arista/avd/docs/plugins/Modules and action plugins/validate_and_template.md [hunk 49]
```diff
@@ -0,0 +1,56 @@
+# validate_and_template
+
+!!! note
+    Always use the FQCN (Fully Qualified Collection Name) `arista.avd.validate_and_template` when using this plugin.
+
+Validate input data according to Schema\, render Jinja2 template and write result to a file\.
+
+## Synopsis
+
+The \`arista\.avd\.validate\_and\_template\` Action Plugin performs data conversions and validation according to the supplied Schema\.
```
Comment (JulioPDX) [inline ansible_collections/arista/avd/docs/plugins/Modules and action plugins/validate_and_template.md]: Present in multiple docs, I'm debating if the synopsis should be one main paragraph vs a row of sentences.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/plugins/Modules and action plugins/validate_and_template.md]: If the description is a list in the source files, it will be renderes as seperate paragraphs. So we should change the source files to use multiline strings if we want the output to be a single paragraph.

File: ansible_collections/arista/avd/docs/plugins/Filter plugins/decrypt.md (hunk 45 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/plugins/Filter plugins/decrypt.md:20]: I think I got rid of them now.

File: ansible_collections/arista/avd/docs/plugins/Filter plugins/range_expand.md (hunk 46 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/plugins/Filter plugins/range_expand.md:44]: No this one is a list input where the first one is a string.

File: ansible_collections/arista/avd/docs/plugins/Modules and action plugins/eos_designs_structured_config.md (hunk 47 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/plugins/Modules and action plugins/eos_designs_structured_config.md]: fixed

File: ansible_collections/arista/avd/docs/plugins/Modules and action plugins/inventory_to_container.md (hunk 48 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/plugins/Modules and action plugins/inventory_to_container.md]: fixed

File: ansible_collections/arista/avd/docs/templates/plugin-docs.j2 [hunk 50]
```diff
@@ -0,0 +1,163 @@
...
+ Copyright (c) 2023 Arista Networks, Inc.
+ Use of this source code is governed by the Apache License 2.0
+ that can be found in the LICENSE file.
+#}
+{# Based on https://github.com/xlab-steampunk/ansible-doc-extractor/blob/master/src/ansible_doc_extractor/templates/module.md.j2 #}
+<!--
+  ~ Copyright (c) 2023 Arista Networks, Inc.
+  ~ Use of this source code is governed by the Apache License 2.0
+  ~ that can be found in the LICENSE file.
+  -->
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/templates/plugin-docs.j2:11]: so... do we need copyright in auto-generated files? ... :) (we can approve like this in any case, it just seems overkill)

**Review by gmuloc — APPROVED** [review]
Thanks for this - tedious - but amazing changes :)

File: ansible_collections/arista/avd/docs/templates/plugin-docs.j2 (hunk 50 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/templates/plugin-docs.j2:11]: Yes, when the files are stored in our repo, we do.

Comment (gmuloc) [reply ansible_collections/arista/avd/docs/templates/plugin-docs.j2:11]: if we go to this extent we could add a line as comment to say "this has been auto-generated, do not edit"

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/templates/plugin-docs.j2:11]: It will be removed again once we get the mkdocs diff stuff sorted out.

**Review by carlbuchmann — APPROVED** [review]
LGTM
